what4/src/What4/Expr/Builder.hs

```diff
@@ -1639,11 +1640,13 @@ instance IsExprBuilder (ExprBuilder t st fs) where
 
   getAnnotation _sym e =
     case e of
+      BoundVarExpr (bvarId -> n) -> Just n
       NonceAppExpr (nonceExprApp -> Annotation _ n _) -> Just n
       _ -> Nothing
 
   getUnannotatedTerm _sym e =
     case e of
+      BoundVarExpr {} -> Just e
```

Comment:
This is a bit of a strange thing to do, since the expression that we returning here _is_ annotated (with the same annotation as the argument expression, no less). I suppose I imagined that `BoundVarExpr` had an underlying `Expr` to return, but upon closer inspection, it doesn't. Hm.

In light of this, perhaps `getUnannotatedTerm` should return `Nothing` for `BoundVarExpr`. We should also amend the Haddocks for `getUnannotatedTerm` to mention that returning `Nothing` doesn't imply that the argument expression is completely unannotated, but rather that it is not feasible to separate the expression from the annotation. If we do this, then I would be less bothered by the fact that `getUnannotatedTerm` has fewer special cases than `annotateTerm`/`getAnnotation` do.

Reply:
Removed, documented!